### PR TITLE
Migrate verify case id in statusdb

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description
The function: `verify_case_id_in_statusdb` belongs to the Analysis API, but has nothing to with an analysis. This function belongs to the status database API as it only involves reading the database and handling the result of the query.

### Added

- Tests

### Changed

- Refactor and move `verify_case_id_in_statusdb` to `verify_case_exists` from Analysis API to Status DB

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b migrate_verify_case_id_in_statusdb -a
    ```

### How to test

- [x] See below

### Expected test outcome

- [x] Check that the case is found using the Analysis API (via status DB)
- [x] Take a screenshot and attach or copy/paste the output.

## Review

- [x] Tests executed by HS
- [x] "Merge and deploy" approved by @Vince-janv 
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/store/api/find/test_find_business_data.py [hunk 1]
```diff
@@ -516,6 +516,60 @@ def test_find_cases_for_non_existing_case(store_with_multiple_cases_and_samples:
...
+    # WHEN validating if the case exists
+    store_with_multiple_cases_and_samples.verify_case_exists(
+        case_internal_id=case_id_with_multiple_samples
+    )
+
+    # THEN the case is found
+    assert f"Case {case_id_with_multiple_samples} exists in Status DB" in caplog.text
+
+
+def test_verify_case_exists_with_none_existing_case(
```
Comment (Vince-janv) [inline tests/store/api/find/test_find_business_data.py:537]:
```suggestion
def test_verify_case_exists_with_non_existing_case(
```

**Review by Vince-janv — APPROVED** [review]
Nice 👍

**Review by diitaz93 — APPROVED** [review]
👍

**Review by diitaz93 — APPROVED** [review]
👍

File: tests/store/api/find/test_find_business_data.py (hunk 1 above)
Comment (henrikstranneheim) [reply tests/store/api/find/test_find_business_data.py]: Done

**Comment by henrikstranneheim** [comment]
### Starting MIP ✅ 

```
θ67° [henrik.stranneheim@hasta:~] [S_base] 50s $ cg workflow mip-dna start -d -qos low  --mip-dry-run justhusky
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Case justhusky exists in Status DB
Starting full MIP analysis workflow for case justhusky
```

**Comment by henrikstranneheim** [comment]
Deployed on Hasta:

```
Log deploy... done.
cg, version 31.0.2
θ68° [henrik.stranneheim@hasta:~] [P_base] 3m16s $
```

**Comment by henrikstranneheim** [comment]
@sofstam I think you need to update #1998 to accommodate these changes.

**Comment by sofstam** [comment]
Thanks @henrikstranneheim, I will :)

**Pull request merged**